### DISCUSSION TITLE
Steps to create a quarto journal extension

### DISCUSSION THREAD

**Discussion by eteitelbaum** [quarto-dev/quarto-cli#5063] [Q&A]
Hello,! I am trying to get started on making a Quarto journal extension. I am struggling with a few things after having looked at a few examples of templates on GitHub and after reading the many pages discussing templates and extensions on the Quarto site: 

1. How do I start? I see two methods being discussed. One discussed in the [Journal Formats](https://quarto.org/docs/journals/formats.html) quick start section says to type `quarto create extension journal` in the terminal. But then in the Quarto journals [GitHub repo,](https://github.com/quarto-journals/article-format-template) a different workflow is identified where you are encouraged to click "Use this Template" to create a new repo. Are these equivalent workflows. Which one is better? 

2. This article on [article templates](https://quarto.org/docs/journals/templates.html) identifies two ways of developing a new article format. One is to selectively replace partials in the master Quarto LaTeX or HTML template. The other is to replace the entire template and then add back partials provided with Quarto. [Here](https://www.overleaf.com/project/6428681caf8eba9be77e32e8) is the LaTeX template from the American Political Science Association that I would like to do in Quarto. Which of these two methods would be the best method for porting this into Quarto? 

3. I am wondering about document class options, like whether the document is single or double spaced, whether it is blinded or not, whether it includes a word count, e.g. `\documentclass[bibtex,autowc]{apsr_submission}`. Some journals have different templates available based on whether it is under review or in the preprint stage, e.g. `\documentclass[preprint,journal]{vgtc}`. Is there a way to get these options to show up as parameters in the YAML header of the template? 

Thanks!

**Comment by baptiste** [comment]
We don't have permission to view the Overleaf project.

Generally speaking, 1) and 2) are a matter of taste, and will also depend on the template you want to port to Quarto. If it's quite similar to another Quarto extension, it makes a lot of sense to start from that repo and make the adjustments, rather than starting from scratch. But some people might prefer to start with a clean slate and be in control of everything they add, piece by piece, rather than starting from someone else's files.

It's a little similar with 2), in that both approaches are possible. With the custom template you get full control over what's in it, line by line, whereas replacing only partials makes you follow the built-in templates more closely. The trade-off here is between convenience (the Quarto template supports all Quarto features, so you don't have to think about implementing/importing them, and they're all documented), and control – if you find that the Quarto/pandoc template is not to your taste (either because it requires extensive additions for the functionality you need, or because it defines many features that you don't want/need – or both). 

Pragmatically, you might want to start with the easier workflow (fork a template from a similar journal extension), and use partials rather than a full template, and if later on you find that there's too much complexity in the code, you'll be in a good position to know what you need and trim it down to a more minimalist approach. 
Just my personal opinion though!

**Reply by eteitelbaum** [reply]
Yes now I realize that you have to log in to see the template on overleaf. I have created a temporary repo where I have [posted](https://github.com/eteitelbaum/apsr-quarto-test/blob/main/_extensions/apsr/apsr.tex) the file. I am trying to figure out if it makes more sense to take the various elements of the LaTeX template and construct the Quarto extension with partials or if I should try to do it with a .tex template in the `_extension.yml` file like this: 

``` 
pdf:
  template: apsr.tex
```

I feel like there might be too much going on in the APSR LaTeX template to make that approach work and am leaning towards using partials.

**Comment by cscheid** [comment]
> `\documentclass[preprint,journal]{vgtc}`

I wouldn't have brought it up had I not seen this specific line, but:  https://github.com/cscheid/quarto-ieee-tvcg (I haven't touched it in a while, but I'm planning on doing so in the near future!)

**Reply by eteitelbaum** [reply]
Yes I've been working with your template!
